<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BitmapBinarizationClass</class>
 <widget class="QMainWindow" name="BitmapBinarizationClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>882</width>
    <height>652</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BitmapBinarization</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="inputImage">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>400</width>
      <height>350</height>
     </rect>
    </property>
    <property name="text">
     <string>                                                     Image before</string>
    </property>
   </widget>
   <widget class="QLabel" name="outputImage">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>10</y>
      <width>400</width>
      <height>350</height>
     </rect>
    </property>
    <property name="text">
     <string>                                                           Image after</string>
    </property>
   </widget>
   <widget class="QSlider" name="thresholdSlider">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>430</y>
      <width>481</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="threadsSlider">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>490</y>
      <width>481</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>64</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="inputButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>370</y>
      <width>75</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Input</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>530</y>
      <width>91</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string>Execution time:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>410</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Binarization threshold:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>470</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of threads:</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="cppRadioButton">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>560</y>
      <width>89</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>C++</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="asmRadioButton">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>560</y>
      <width>89</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>ASM</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>470</y>
      <width>211</width>
      <height>121</height>
     </rect>
    </property>
    <property name="text">
     <string>START</string>
    </property>
   </widget>
   <widget class="QLabel" name="lbl_threshold">
    <property name="geometry">
     <rect>
      <x>135</x>
      <y>410</y>
      <width>49</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="lbl_threads">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>470</y>
      <width>49</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>1</string>
    </property>
   </widget>
   <widget class="QLabel" name="lbl_exeTime">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>530</y>
      <width>91</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="lbl_background">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>-10</y>
      <width>881</width>
      <height>661</height>
     </rect>
    </property>
    <property name="text">
     <string>TextLabel</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>380</y>
      <width>75</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Show details</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>360</y>
      <width>181</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Show histogram of input image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_4">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>360</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Show histogram of output image</string>
    </property>
   </widget>
   <zorder>lbl_background</zorder>
   <zorder>inputImage</zorder>
   <zorder>outputImage</zorder>
   <zorder>thresholdSlider</zorder>
   <zorder>threadsSlider</zorder>
   <zorder>inputButton</zorder>
   <zorder>label_3</zorder>
   <zorder>label_4</zorder>
   <zorder>label_5</zorder>
   <zorder>cppRadioButton</zorder>
   <zorder>asmRadioButton</zorder>
   <zorder>pushButton_3</zorder>
   <zorder>lbl_threshold</zorder>
   <zorder>lbl_threads</zorder>
   <zorder>lbl_exeTime</zorder>
   <zorder>pushButton_2</zorder>
   <zorder>pushButton</zorder>
   <zorder>pushButton_4</zorder>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="BitmapBinarization.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>inputButton</sender>
   <signal>clicked()</signal>
   <receiver>BitmapBinarizationClass</receiver>
   <slot>inputButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>421</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>382</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>thresholdSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BitmapBinarizationClass</receiver>
   <slot>thresholdSliderValueChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>269</x>
     <y>467</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>440</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>threadsSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BitmapBinarizationClass</receiver>
   <slot>threadsSliderValueChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>370</x>
     <y>531</y>
    </hint>
    <hint type="destinationlabel">
     <x>504</x>
     <y>505</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>BitmapBinarizationClass</receiver>
   <slot>startButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>693</x>
     <y>581</y>
    </hint>
    <hint type="destinationlabel">
     <x>605</x>
     <y>594</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>BitmapBinarizationClass</receiver>
   <slot>detailsButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>804</x>
     <y>396</y>
    </hint>
    <hint type="destinationlabel">
     <x>909</x>
     <y>423</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>BitmapBinarizationClass</receiver>
   <slot>histogramButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>445</x>
     <y>381</y>
    </hint>
    <hint type="destinationlabel">
     <x>133</x>
     <y>651</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>BitmapBinarizationClass</receiver>
   <slot>pieChartButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>569</x>
     <y>377</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>662</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>inputButtonClicked()</slot>
  <slot>outputButtonClicked()</slot>
  <slot>thresholdSliderValueChanged()</slot>
  <slot>threadsSliderValueChanged()</slot>
  <slot>debugButtonClicked()</slot>
  <slot>startButtonClicked()</slot>
  <slot>detailsButtonClicked()</slot>
  <slot>histogramButtonClicked()</slot>
  <slot>pieChartButtonClicked()</slot>
 </slots>
</ui>
